velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -69,6 +61,16 @@ NestedLoopJoinProbe::NestedLoopJoinProbe(
   }
 }
 
+bool NestedLoopJoinProbe::needsProbeMismatch(core::JoinType joinType) {
```

Comment:
@karteekmurthys : Another option for you is to add (with or) joinCondition == nullptr to this condition that returns nullptr in buildMismatchedOutput https://github.com/facebookincubator/velox/blob/main/velox/exec/NestedLoopJoinProbe.cpp#L225

I feel that is a bit cleaner than changing needsProbe(Build)Mismatch here.

Can you try that ?